### PR TITLE
Port Javascript test suite from Teaspoon/QUnit to Jasmine

### PR DESCRIPTION
**Pull request by kevindew**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

This project was inconsistent [with other GOV.UK Rails applications](https://docs.publishing.service.gov.uk/manual/conventions-for-rails-applications.html#testing-javascript-code) by using [Teaspoon/QUnit for JavaScript testing](https://github.com/jejacks0n/teaspoon) rather than Jasmine. 

Teaspoon is quite an old utility to run JavaScript tests through Rails, it bundles JS test frameworks, such as QUnit and Jasmine. However in recent years it's quite far behind the latest versions of these frameworks as it has become less common to use Rails gems for Javascript packages. 

This PR might look quite scary but it takes a very modest approach to porting the tests over with very little changes to what is actually being tested (it also removes some dead code). It's easiest reviewed commit by commit.

A motivation for this is that the latest version of Teaspoon doesn't work with this app anymore (see: https://github.com/alphagov/whitehall/commit/58c2e66c0c81a6112844b72c1cc2dca20ddd6bcf)

### CODE REVIEW

File: spec/javascripts/admin/views/editions/_form.spec.js [hunk 1]
```diff
@@ -1,512 +1,442 @@
...
-module('admin-edition-form-first-published-at: ', {
-  setup: function () {
-    $('#qunit-fixture').append(form)
-    $('#qunit-fixture form').append(firstPublishedAtFieldset)
-
-    GOVUK.adminEditionsForm.init({
-      selector: 'form#non-english',
-      right_to_left_locales: ['ar']
+    it('hides and resets the locale input when foreign language only is unchecked', function () {
+      form.find('#create_foreign_language_only').click().click()
```
Comment (sihugh) [inline spec/javascripts/admin/views/editions/_form.spec.js:33]: duplicate `click()` here?

File: spec/javascripts/admin/views/editions/_form.spec.js [hunk 2]
```diff
@@ -1,512 +1,442 @@
...
-
-test("selecting the 'World news story' News Article type hides the role-appointments fieldset and resets the role-appointments", function () {
-  var $select = $('select#edition_news_article_type_id')
-
-  $('#edition_role_appointment_ids').val('3850').change()
 
-  $select.find('option:contains(World news story)').prop('selected', true).change()
-  ok($('fieldset.role-appointments').is(':hidden'), 'role-appointments fieldset is visible')
+    it('shows the foreign language options when a "World news story" is selected', function () {
+      form.find('#edition_news_article_type_id').val('2').change() // Press release
```
Comment (sihugh) [inline spec/javascripts/admin/views/editions/_form.spec.js:62]: I'm going to test you on your knowledge of these numbers at some point :-)

File: spec/support/jasmine-browser.json [hunk 3]
```diff
@@ -8,6 +8,7 @@
     "**/*[sS]pec.js"
   ],
   "helpers": [
+    "https://unpkg.com/jasmine-ajax@4.0.0/lib/mock-ajax.js",
```
Comment (sihugh) [inline spec/support/jasmine-browser.json:11]: Should we vendor this?

**Review by sihugh — COMMENTED** [review]
I've got as far as "Port statistics_announcement_dates/_form JS tests from QUnit to Jasmine" and it looks OK so far

File: spec/javascripts/admin/views/editions/_form.spec.js (hunk 1 above)
Comment (kevindew) [reply spec/javascripts/admin/views/editions/_form.spec.js]: It is intentional, as it's to be clicked on and then clicked off. But I can understand that it is not clear to read, so I've merged this into the test above to try make it clear.

File: spec/support/jasmine-browser.json (hunk 3 above)
Comment (kevindew) [reply spec/support/jasmine-browser.json:11]: We've been experimenting with the CDN approach on Content Publisher before: https://github.com/alphagov/content-publisher/blob/3f14860c12aae112f0da2afb5314740d253177d2/spec/support/jasmine-browser.json 

It seems a bit cleaner than vendoring.

Ideally we'd pull these through with an NPM module but we don't know how to do that yet.

Would some more info in the commit message help?

**Review by benjamineskola — APPROVED** [review]

File: spec/support/jasmine-browser.json (hunk 3 above)
Comment (sihugh) [reply spec/support/jasmine-browser.json:11]: It's fine 👍

**Review by sihugh — APPROVED** [review]

**Pull request merged**